tests/testthat/test-check.R

```diff
@@ -0,0 +1,46 @@
+test_that("assert_character_param", {
+  expect_no_error(assert_character_param("name", "value", len = 1))
+  expect_error(assert_character_param("name", c("value", "value"), len = 1))
+  expect_no_error(assert_character_param("name", c("value1", "value2")))
+  expect_error(assert_character_param("name", 5))
+  expect_error(assert_character_param("name", NULL))
+  expect_no_error(assert_character_param("name", NULL, required = FALSE))
+  expect_error(assert_character_param("name", list("value1", 5)))
+})
+
+test_that("assert_integerish_param", {
+  expect_no_error(assert_integerish_param("name", 5, len = 1))
+  expect_error(assert_integerish_param("name", c(5, 6), len = 1))
+  expect_no_error(assert_integerish_param("name", c(5, 6)))
+  expect_error(assert_integerish_param("name", "value", len = 1))
+  expect_error(assert_integerish_param("name", NULL, len = 1))
+  expect_no_error(assert_integerish_param("name", NULL, len = 1, required = FALSE))
+  expect_error(assert_integerish_param("name", list(5, 6)))
+})
+
+test_that("assert_date_param", {
+  expect_no_error(assert_date_param("name", "2020-01-01", len = 1))
+  expect_error(assert_date_param("name", c("2020-01-01", "2021-01-02"), len = 1))
+  expect_no_error(assert_date_param("name", c("2020-01-01", "2021-01-02")))
+  expect_no_error(assert_date_param("name", c(20200101, 20200101)))
+  expect_no_error(assert_date_param("name", c(as.Date("2020-01-01"), as.Date("2020-01-02"))))
+  expect_error(assert_date_param("name", NULL))
+  expect_no_error(assert_date_param("name", NULL, required = FALSE))
+  expect_error(assert_date_param("name", list(20200101, 20200101)))
+  expect_no_error(assert_date_param("name", c(as.Date("2020-01-01"), 20200102, "20200103")))
+  expect_no_error(assert_date_param("name", "*"))
+})
+
+test_that("assert_timeset_param", {
+  expect_no_error(assert_timeset_param("name", "2020-01-01", len = 1))
+  expect_error(assert_timeset_param("name", c("2020-01-01", "2021-01-02"), len = 1))
+  expect_no_error(assert_timeset_param("name", c("2020-01-01", "2021-01-02")))
+  expect_no_error(assert_timeset_param("name", c(20200101, 20200101)))
+  expect_no_error(assert_timeset_param("name", c(as.Date("2020-01-01"), as.Date("2020-01-02"))))
+  expect_no_error(assert_timeset_param("name", epirange(20200101, 20200102)))
+  expect_error(assert_timeset_param("name", c(epirange(20200101, 20200102), epirange(20200103, 20200104))))
```

Comment:
issue: we don't want to allow this.  Right now our EpiRange class doesn't react well inside `c`, and I believe this would actually end up asking for 4 dates individually, not 2 ranges.  The list-of-epiranges is currently the way to go.  We might be able to make things more robust with a more featureful EpiRange class, but it's seems like a lot of work for something that seems like an edge case.  Might prefer just banning lists altogether (but keeping Date vectors and single EpiRanges as options).

Reply:
#97